### PR TITLE
feat(util): added new check method: auditLogParts can be only in last part of a chained rule

### PR DESCRIPTION
**Pull request by airween**
See comment under issue [#2530](https://github.com/coreruleset/coreruleset/issues/2530#issuecomment-1140264655)

### CODE REVIEW

**Comment by airween** [comment]
Thanks @RedXanadu for grammatical check :)

File: util/crs-rules-check/README.md [hunk 1]
```diff
@@ -212,3 +212,37 @@ Config file: examples/test5.conf
...
+    t:none,\
+    ctl:auditLogParts=+E,\
+    chain"
+    SecRule RESPONSE_BODY "@rx (?i:JET Database Engine|Access Database Engine|\[Microsoft\]\[ODBC Microsoft Access Driver\])" \
+        "capture,\
+        setvar:'tx.outbound_anomaly_score_pl1=+%{tx.critical_anomaly_score}',\
+        setvar:'tx.sql_injection_score=+%{tx.critical_anomaly_score}'"
+```
+
+In this rule, the `ctl:autidLogParts=+E` is in wrong place, because some non-disruptive actions will be executed on non-disruptive rules (most CRS rules are non-disruptive) even if the chained rules are not satisfied.
```
Comment (fzipi) [inline util/crs-rules-check/README.md:233]: There is a typo here, `autidLogParts` -> `auditLogParts`.

File: util/crs-rules-check/rules-check.py [hunk 2]
```diff
@@ -339,4 +382,17 @@ def msg(msg):
                 errmsgf(e)
             errmsg(d.strip("\n"))
 
+        ### check `ctl:auditLogParts=+E` right place in chained rules
+        c.check_ctl_audit_log()
+        if len(c.auditlogparts) == 0:
+            msg(" 'ctl:auditLogParts' actions are in right place.")
```
Comment (fzipi) [inline util/crs-rules-check/rules-check.py:388]: Maybe `'ctl:auditLogParts' is in the last rule of the chain` ?

File: util/crs-rules-check/README.md (hunk 1 above)
Comment (airween) [reply util/crs-rules-check/README.md]: Thanks, fixed.

File: util/crs-rules-check/rules-check.py [hunk 3]
```diff
@@ -339,4 +382,17 @@ def msg(msg):
...
+        else:
+            errmsg(" Found 'ctl:auditLogParts' action is in wrong place.")
```
[the remark this answers is not in the document]
Comment (airween) [reply util/crs-rules-check/rules-check.py:390]: Good question. First message (`Found 'ctl:auditLogParts' action is in wrong place.`) is a "group" text, eg:

```
Config file: ../../rules/REQUEST-943-APPLICATION-ATTACK-SESSION-FIXATION.conf
 Parsing ok.
 Ignore case check ok.
 Action order check ok.
 Indentation check ok.
 Found 'ctl:auditLogParts' action is in wrong place.
  file=../../rules/REQUEST-943-APPLICATION-ATTACK-SESSION-FIXATION.conf, line=68, endLine=68, title='ctl:auditLogParts' action in wrong place: action can only be placed in last part of a chained rule (rule: 943110)
```
The second one is a detailed message. May be this help to see the difference.

File: util/crs-rules-check/rules-check.py (hunk 2 above)
Comment (airween) [reply util/crs-rules-check/rules-check.py:388]: Same as above - I do not know, what would be the better. Note, that I asked @RedXanadu for help to make messages :).

Comment (airween) [reply util/crs-rules-check/rules-check.py:388]: Sorry, an example here too:

```
Config file: ../../rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf
 Parsing ok.
 Ignore case check ok.
 Action order check ok.
 Indentation check ok.
 'ctl:auditLogParts' actions are in right place.
```

**Review by fzipi — APPROVED** [review]
Then everything is cool.

**Comment by fzipi** [comment]
Merging after approval.

**Pull request merged**